osu.Game/Screens/OnlinePlay/Lounge/Components/RoomPanel.cs

```diff
@@ -556,5 +555,57 @@ private void load(OsuColour colours)
                 };
             }
         }
+
+        public partial class RoomNameLine : FillFlowContainer
+        {
+            private readonly string? roomUrl;
+            private readonly bool showExternalLink;
+
+            private TruncatingSpriteText spriteText = null!;
+            private ExternalLinkButton link = null!;
+
+            public LocalisableString Text
+            {
+                get => spriteText.Text;
+                set => spriteText.Text = value;
+            }
+
+            public RoomNameLine(string? roomUrl, bool showExternalLink)
```

Comment:
URL should be non-nullable, or it should be nullable and the external link button shouldn't be added when null, or some other combination. Having these two arguments with two methods of "not working" which result in different behaviours is weird.